src/components/utils/events.ts

```diff
@@ -1,31 +1,34 @@
 import { isEmpty } from '../utils';
 
 /**
- * @class EventDispatcher
+ * Mapped type with subscriptions list
  *
- * Has two important methods:
+ * event name -> array of callbacks
+ */
+type Subscriptions<EventMap> = {
+  [name in keyof EventMap]: Array<(data?: EventMap[name]) => unknown>
+};
+
+/**
+ * Provides methods for working with Event Bus:
  *    - {Function} on - appends subscriber to the event. If event doesn't exist - creates new one
  *    - {Function} emit - fires all subscribers with data
  *    - {Function off - unsubscribes callback
- * @version 1.0.0
- * @typedef {Events} Events
- * @property {object} subscribers - all subscribers grouped by event name
  */
-export default class EventsDispatcher<Events extends string = string> {
+export default class EventsDispatcher<EventMap extends Record<string, unknown> = undefined> {
   /**
+   * All subscribers grouped by event name
    * Object with events` names as key and array of callback functions as value
-   *
-   * @type {{}}
    */
-  private subscribers: {[name: string]: Array<(data?: object) => unknown>} = {};
+  private subscribers: Subscriptions<EventMap> = <Subscriptions<EventMap>>{};
 
   /**
    * Subscribe any event on callback
    *
-   * @param {string} eventName - event name
-   * @param {Function} callback - subscriber
+   * @param eventName - event name
+   * @param callback - subscriber
    */
-  public on(eventName: Events, callback: (data: object) => unknown): void {
+  public on<Name extends keyof EventMap>(eventName: Name, callback: (data: EventMap[Name]) => unknown): void {
```

Comment:
`void` type is better to indicate *you don't expect any return value*

```suggestion
  public on<Name extends keyof EventMap>(eventName: Name, callback: (data: EventMap[Name]) => void): void {
```